### PR TITLE
com.helix_editor.Helix

### PR DESCRIPTION
**Pull request by blt-r**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/helix-editor/helix/issues/2076
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
(Its actually [helix-editor.com](https://helix-editor.com), but dashes are supposed to be replaced with underscores)
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64", "aarch64"]
```
Comment (barthalion) [inline flathub.json:2]: No need to define that, these are the only supported architectures already.

**Comment by barthalion** [comment]
bot, build com.helix_editor.Helix

File: helix.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  <screenshots>
+    <screenshot type="default">
+      <caption>Helix with default theme</caption>
+      <image>https://github.com/helix-editor/helix/raw/d4565b4404cabc522bd60822abd374755581d751/screenshot.png</image>
+    </screenshot>
+  </screenshots>
+
+  <url type="homepage">https://helix-editor.com/</url>
+
+  <content_rating>none</content_rating>
```
Comment (barthalion) [inline helix.appdata.xml:33]: This is wrong. It should be `  <content_rating type="oars-1.1" />`.

File: flathub.json (hunk 1 above)
Comment (blt-r) [reply flathub.json]: Ok, removed it

File: helix.appdata.xml (hunk 2 above)
Comment (blt-r) [reply helix.appdata.xml]: I see. Weird that `org.freedesktop.appstream-glib validate` didn't complain. Anyway, its fixed

File: com.helix_editor.Helix.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        sources:
+          - type: archive
+            url: https://www.x.org/releases/individual/lib/libXmu-1.1.3.tar.bz2
+            sha256: 9c343225e7c3dc0904f2122b562278da5fed639b1b5e880d25111561bac5b731
+        cleanup:
+          - /include
+          - /lib/pkgconfig
+          - '*.la'
+
+finish-args:
```
Comment (hfiguiere) [inline com.helix_editor.Helix.yml:77]: finish args are usually above modules.

File: com.helix_editor.Helix.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - --env=HELIX_RUNTIME=/app/lib/helix/runtime
+  # Filesystem acess
+  - --filesystem=host
```
Comment (hfiguiere) [inline com.helix_editor.Helix.yml:80]: this is a very broad filesystem access.

File: com.helix_editor.Helix.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      - install -D hx /app/bin/hx
+
+      - mkdir -p /app/lib/helix
+      - rm -rf /app/lib/helix/runtime
+      - cp -r runtime /app/lib/helix/runtime
+
+      - install -D Helix.desktop /app/share/applications/com.helix_editor.Helix.desktop
+      - install -D helix.png /app/share/icons/hicolor/256x256/apps/com.helix_editor.Helix.png
+
+      - install -D helix.appdata.xml /app/share/metainfo/com.helix_editor.Helix.metainfo.xml
```
Comment (hfiguiere) [inline com.helix_editor.Helix.yml:22]: these files should be submitted upstream. A desktop app without icon or .desktop is just a bug.

File: com.helix_editor.Helix.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
+app-id: com.helix_editor.Helix
+
+runtime: org.freedesktop.Sdk
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+
+command: hx
+
+modules:
+  - name: hx
```
Comment (hfiguiere) [inline com.helix_editor.Helix.yml:10]: this being the main module. it should be last.

also any chance to build from source?

Comment (blt-r) [reply com.helix_editor.Helix.yml:10]: Might as well build from source, no problem with that. I don't know what are the best practices here, if building from source is better then I'll do that, but I think it's easier to just use prebuilt things

File: com.helix_editor.Helix.yml (hunk 5 above)
Comment (blt-r) [reply com.helix_editor.Helix.yml:22]: Icon and .desktop are from upstream (https://github.com/helix-editor/helix/tree/master/contrib), but they don't put them in release archive. The appdata manifest however I wrote myself

File: com.helix_editor.Helix.yml (hunk 3 above)
Comment (blt-r) [reply com.helix_editor.Helix.yml]: Fixed it

**Comment by blt-r** [comment]
I actually just realized we also need `--talk-name=org.freedesktop.Flatpak` so we can `flatpak-spawn` language servers installed on the host system. Is it ok if I add that?

File: com.helix_editor.Helix.yml (hunk 6 above)
Comment (hfiguiere) [reply com.helix_editor.Helix.yml]: building from source is always preferred.

**Comment by barthalion** [comment]
It's allowed for IDEs/text editors, but you need to submit a PR to https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json.

File: com.helix_editor.Helix.yml (hunk 6 above)
Comment (blt-r) [reply com.helix_editor.Helix.yml]: Ok so I tried to make it build from source, but I don't think it's worth it. We need to build tree-sitter grammars. And for that we need to fetch sources for them. So we will have to fetch pkgs.tspm.io for sources and generate manifest module source definitions from them, and then compile them. It just better to use prebuilt grammars. But if we use prebuilt grammars, we might as well use the prebuilt binary

**Comment by barthalion** [comment]
bot, build com.helix_editor.Helix

**Pull request closed without merging**